src/box/sql/build.c

```diff
@@ -2902,14 +2868,28 @@ sql_drop_index(struct Parse *parse_context)
 		}
 		goto exit_drop_index;
 	}
-	index_name = sql_name_from_token(&drop_def->name);
+	uint32_t index_id = sql_index_id_by_token(space, &drop_def->name);
+	if (index_id == UINT32_MAX) {
+		if (if_exists)
+			goto exit_drop_index;
+		const char *name_str = sql_tt_name_from_token(&drop_def->name);
+		diag_set(ClientError, ER_NO_SUCH_INDEX_NAME, name_str,
+			 space->def->name);
+		parse_context->is_aborted = true;
+		goto exit_drop_index;
+	}
 
-	vdbe_emit_index_drop(parse_context, index_name, space->def,
-			     ER_NO_SUCH_INDEX_NAME, if_exists);
+	int regs = sqlGetTempRange(parse_context, 3);
+	sqlVdbeCountChanges(v);
```

Comment:
Side note: Could you please explain, why this call is required?

Reply:
To count tuples addition and removal in system spaces. For example, CREATE TABLE adds one tuple to _space, so it returns `row_count = 1`. The same for DROP TABLE and so on. At the same time ALTER TABLE do not add or remove tuples, so `row_count = 0`.

Reply:
Got it, thanks!